### ISSUE TITLE
mvnd and .mvn

### ISSUE THREAD

**Issue by delanym** [apache/maven-mvnd#813]
I would like the configuration in `.mvn/maven.config` to be used no matter the executable (mvn, mvnw, mvnd).

Currently mvnd (via the new maven wrapper type=only-script) doesn't consider those properties, e.g. `-Dorg.slf4j.simpleLogger.defaultLogLevel=warn`

This is surprising behaviour. There's no mention at https://github.com/apache/maven-mvnd of any files in .mvn/maven.config, jvm.config, extensions.xml so its not clear what the relationship is.

**Comment by gnodet** [comment]
THe `.mvn/maven.config` should be parsed.  However, `mvnd` does not use the slf4j simple logger, but uses logback.  So that may be the reason why you don't see any effect when passing your system property.

**Comment by delanym** [comment]
Ok, so add `-Dlogback.configurationFile=​logback.​xml` to `maven.config`? Will that always find `.mvn/logback.xml`?

**Comment by gnodet** [comment]
Unfortunately, the `.mvn/maven.config` file does not support any kind of interpolation so the file is most probably resolved relatively to the current directory.

I think this is related to https://issues.apache.org/jira/browse/MNG-5930, https://issues.apache.org/jira/browse/MNG-6303, https://issues.apache.org/jira/browse/MNG-6762 ...

**Comment by michael-o** [comment]
> Ok, so add `-Dlogback.configurationFile=​logback.​xml` to `maven.config`? Will that always find `.mvn/logback.xml`?

This is wrong! What you want is a system property, but `maven.config` is *only* about Maven user properties. You must use `jvm.config` for this.

**Comment by gnodet** [comment]
> > Ok, so add `-Dlogback.configurationFile=​logback.​xml` to `maven.config`? Will that always find `.mvn/logback.xml`?
> 
> This is wrong! What you want is a system property, but `maven.config` is _only_ about Maven user properties. You must use `jvm.config` for this.

That's irrelevant.  The question is the same with `jvm.config`...  Or that mean that there's a need for yet another level of property, which may become a bit too much...

**Comment by michael-o** [comment]
> > > Ok, so add `-Dlogback.configurationFile=​logback.​xml` to `maven.config`? Will that always find `.mvn/logback.xml`?
> > 
> > 
> > This is wrong! What you want is a system property, but `maven.config` is _only_ about Maven user properties. You must use `jvm.config` for this.
> 
> That's irrelevant. The question is the same with `jvm.config`... Or that mean that there's a need for yet another level of property, which may become a bit too much...

I wouldn't say so because both files are processed in completely different ways. This matters.

**Comment by gnodet** [comment]
> > > > Ok, so add `-Dlogback.configurationFile=​logback.​xml` to `maven.config`? Will that always find `.mvn/logback.xml`?
> > > 
> > > 
> > > This is wrong! What you want is a system property, but `maven.config` is _only_ about Maven user properties. You must use `jvm.config` for this.
> > 
> > 
> > That's irrelevant. The question is the same with `jvm.config`... Or that mean that there's a need for yet another level of property, which may become a bit too much...
> 
> I wouldn't say so because both files are processed in completely different ways. This matters.

This is purely a technical problem however.   Some system properties can be set before the JVM is started (using `-Dxxx` options on the JVM command line), others can be set by the java code using `System.setProperty`.  Both may need to be supported, and splitting completely user properties from system properties (i.e. not setting user properties as system properties) may open the need for properties to be set by maven early (i.e. same time as user properties atm).

**Comment by michael-o** [comment]
> > > > > Ok, so add `-Dlogback.configurationFile=​logback.​xml` to `maven.config`? Will that always find `.mvn/logback.xml`?
> > > > 
> > > > 
> > > > This is wrong! What you want is a system property, but `maven.config` is _only_ about Maven user properties. You must use `jvm.config` for this.
> > > 
> > > 
> > > That's irrelevant. The question is the same with `jvm.config`... Or that mean that there's a need for yet another level of property, which may become a bit too much...
> > 
> > 
> > I wouldn't say so because both files are processed in completely different ways. This matters.
> 
> This is purely a technical problem however. Some system properties can be set before the JVM is started (using `-Dxxx` options on the JVM command line), others can be set by the java code using `System.setProperty`. Both may need to be supported, and splitting completely user properties from system properties (i.e. not setting user properties as system properties) may open the need for properties to be set by maven early (i.e. same time as user properties atm).

Correct. My point was to clarify for/with the user that he is not interacting with system properties actually.

**Comment by delanym** [comment]
No special requirements: `.mvn/mvnd.properties` file will do fine to hold `-Dmvnd.logback=logback.xml`. Although I would expect an error or at least a warning when the file is missing?
